From dfc7f3d72b99a98f04ba10d986382059ce34c958 Mon Sep 17 00:00:00 2001
From: Kyle Willmon <kylewillmon@gmail.com>
Date: Thu, 22 Sep 2011 15:03:02 -0500
Subject: [PATCH] Better error message for dpkg-gencontrol

dpkg-gencontrol now gives a descriptive error message when no binary
packages are found in debian/control
---
 scripts/dpkg-gencontrol.pl |    5 ++++-
 1 files changed, 4 insertions(+), 1 deletions(-)

diff --git a/scripts/dpkg-gencontrol.pl b/scripts/dpkg-gencontrol.pl
index 637d5f5..9682e5c 100755
--- a/scripts/dpkg-gencontrol.pl
+++ b/scripts/dpkg-gencontrol.pl
@@ -152,9 +152,12 @@ if (defined($oppackage)) {
     defined($pkg) || error(_g("package %s not in control info"), $oppackage);
 } else {
     my @packages = map { $_->{'Package'} } $control->get_packages();
-    @packages==1 ||
+    if (@packages > 1) {
         error(_g("must specify package since control info has many (%s)"),
               "@packages");
+    } elsif (@packages < 1) {
+        error(_g("no package sections found in control file"));
+    }
     $pkg = $control->get_pkg_by_idx(1);
 }
 $substvars->set_msg_prefix(sprintf(_g("package %s: "), $pkg->{Package}));
-- 
1.7.5.4

